### PR TITLE
[7.7.x] RHPAM-673 - Ensure only one message is sent at time in the websocket …

### PR DESCRIPTION
**Pull request by cristianonicolai**
…session

### CODE REVIEW

**Comment by cristianonicolai** [comment]
@mswiderski best solution I found so far to make sure a few constraints are met:

- There is only one message sent at time in a session. ( Using single thread executor per session and use sync send )
- Failures to deliver messages do not have an impact across different sessions. ( They are now sent independently if one session is failing, the others will continue to receive messages. )
- Try to keep some consistency for delivering messages in order ( Now it will keep retrying to deliver the same message until it succeeds )

File: kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/notification/WebSocketKieServerControllerNotification.java [hunk 1]
```diff
@@ -41,9 +41,7 @@ public void configure() {
     public void onNotificationClientConnect(final Session session) {
         LOGGER.debug("New Web Socket Notification Client session: {}",
                      session.getId());
-        synchronized (manager) {
```
Comment (mswiderski) [inline kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/notification/WebSocketKieServerControllerNotification.java]: If I am not mistaken we added it here due to some race conditions discovered via cloud testing, @sutaakar am I anywhere close with this...?

Comment (sutaakar) [reply kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/notification/WebSocketKieServerControllerNotification.java]: It was fixing JBPM-6581, the case when several Kie servers connect/disconnect to the controller at once.

Comment (cristianonicolai) [reply kie-server-parent/kie-server-controller/kie-server-controller-websocket/src/main/java/org/kie/server/controller/websocket/notification/WebSocketKieServerControllerNotification.java]: @mswiderski I found that it wasn't needed to sync in here as both add and remove session methods are backed by a concurrent hash map.

**Comment by mswiderski** [comment]
Jenkins retest this

**Comment by mswiderski** [comment]
Jenkins retest this

**Comment by cristianonicolai** [comment]
@mswiderski @sutaakar please take another look.

**Comment by sutaakar** [comment]
looks good to me

**Pull request merged**